### ISSUE TITLE
Scrubbing Non-Crucial Querystrings from SaveKey

### ISSUE THREAD

**Issue by dhaupin** [w8tcha/CKEditor-AutoSave-Plugin#86]
Hi man, sorry for all the issues and stuff the last couple days, but this plugin is great. There is 1 last thing that would make Autosave more epic: the ability to ignore certain URi querystrings so that `SaveKey` is broad enough in all situations. I didnt wanna do a pull request since the querystring function i used is borrowed for a quick and dirty example.

Using CS-Cart, imagine you are editing a product with url like:
`/admin.php?dispatch=products.update&product_id=1121`

After saving the product, the page is reloaded, and the last active tab is opened using url like: `/admin.php?dispatch=products.update&product_id=1121&selected_section=detailed`

Then you change storefronts to some other one, save again, the url changes to something like: `/admin.php?dispatch=products.update&product_id=1121&selected_section=detailed&switch_company_id=1`.

Unfortunately all this makes many different `SaveKey` for the same product, which means that it cant load accurate save points.

**Edit, see referenced pull request https://github.com/w8tcha/CKEditor-AutoSave-Plugin/pull/87**

Thanks man

**Comment by w8tcha** [comment]
Sorry but i think this isn't really a problem!

> After saving the product, the page is reloaded, and the last active tab is opened using url like: /admin.php?dispatch=products.update&product_id=1121&selected_section=detailed

When you set up the plugin correctly, and use the  saveDetectionSelectors with the save button. Then the saved entry is automatically deleted because after you save you dont need to continue store the temp. content.

**Comment by dhaupin** [comment]
I just realized that little hack in config.js only works with 1 instance, so it must go in plugin.js instead. Weird.

Anyways, its a problem IMO because autosave uses local storage names based on url strings that can (will) change in certain platforms. This means it wont load saved data even though it should be. I can think of a few url styles off the top of my head that will cause this very frequently:

- Opencart uses CSRF tokens in url, and they change every admin session, which invalidates autosaved from last session (since &token= will never be the same the next time you log in, wont load data).
- CS-Cart uses last open tab as well as store id switch, which will very often be different from the url string that was autosaved. Example is if you save a product while viewing the "options" tab and then visit that product without viewing a tab (wont load data).
- Other platforms preserve filters from list view so that the same view loads again, or so the next page can use it too. Example is if you filter down a list of products, then click one, perhaps the url will now include search strings for use after save (unless exact same search, wont load data).

What i mean is this: `example.com/index.php?object_id=123` specifies an object to load. Now lets add a bunch of non specifying querystrings that dont do anything in regards to "drilling in" or loading that object `example.com/index.php?object_id=123&sort=asc&open_tab=images&random_token=1231231231234&coming_from=there`

Even though the only thing needed to save/load the page is `object_id`, removing/changing any one of the `open_tab`, `random_token`, or `coming_from` params will change the savekey, which causes failure of the plugin.

I patched it on this end, just figured others may benefit as wel, esp those who use this with Opencart or CS-Cart. Thanks man
